examples/src/main/scala/org/apache/spark/examples/mllib/AbstractParams.scala

```diff
@@ -38,7 +38,7 @@ abstract class AbstractParams[T: TypeTag] {
    */
   override def toString: String = {
     val tpe = tag.tpe
-    val allAccessors = tpe.declarations.collect {
+    val allAccessors = tpe.decls.collect {
```

Comment:
Yes, so I also took a look at this back when I was doing my initial cleanup, there isn't an easy way to suppress deprecation warnings in Scala like these ones (the answer from the Scala people seem to be either "you can't" or  "you could write a compiler plugin" which isn't a feasible option).

We should definitely do this once we get rid of Scala 2.10 (I created https://issues.apache.org/jira/browse/SPARK-14417 to remind us to do the cleanup once 2.10 is dead).

Personally I think the opportunity to get cleanup using these deprecated APIs without version specific code is maybe part of a good reason why we should consider removing 2.10 support, but that is being discussed more on the mailing list.